### ISSUE TITLE
Positive result eventhough it should not

### ISSUE THREAD

**Issue by Captain-P-Goldfish** [jqno/equalsverifier#596]
I think I got a case that produces a positive result eventhough it should not.

I am using lombok in my classes:

```java
@Getter
@Setter
@Builder
@EqualsAndHashCode
public class SoapConfigForm
{

  private KeystoreForm signatureKeystore;

}
```

```java
@Getter
@Setter
@Builder
@EqualsAndHashCode
public class KeystoreForm
{

  private String id;

  @EqualsAndHashCode.Exclude
  private MultipartFile keystore;

  private String keystorePassword;

  private String alias;

  private String keyPassword;

}
```

now the tests:

The keystore form test produces a valid result as long as I add the line with `.withIgnoredFields("keystore")` and fails if I remove this line. So far everything is fine.

```java
public class KeystoreFormTest
{

  @Test
  public void testEquals()
  {
    EqualsVerifier.forClass(KeystoreForm.class)
                  .suppress(Warning.STRICT_INHERITANCE) // we do not want the equals-method to be final
                  .suppress(Warning.NONFINAL_FIELDS) // the forms are mutable
                  .withIgnoredFields("keystore") // the multipart-file is not part of equals and hashcode
                  .verify();
  }
}
```

now the problem:

I think the following test should fail since I did not tell it that the field "keystore" is not part of the equals-method within the class `KeystoreForm`. Am I missing something here?

```java
public class SoapConfigFormTest
{

  @Test
  public void testEquals()
  {
    EqualsVerifier.forClass(SoapConfigForm.class)
                  .suppress(Warning.STRICT_INHERITANCE) // we do not want the equals-method to be final
                  .suppress(Warning.NONFINAL_FIELDS) // the forms are mutable
                  .verify();
  }
}
```

**Comment by jqno** [comment]
When testing a class, EqualsVerifier only needs to generate two different values for its fields, which it uses to see if `equals` returns the correct values. As long as these two values are not equal to each other, everything will be fine. I guess in case of `KeystoreForm`, they are indeed different. If, for some reason, the values generated by EqualsVerifier happen to be equal, you'll get an error message asking to supply prefab values, but in this case that isn't needed.

EqualsVerifier doesn't transitively check `equals` for the field's classes, you'll need to invoke it separately. You indeed do that for `KeystoreForm`, and there it does notice the excluded field.

In other words, it works exaclty as it's supposed to :).
